R/vars_manager.R

```diff
@@ -478,31 +574,43 @@ get_var_details <- function(
             obj, name, no_preview, page_size, page, sort_by, ascending, filters
         )
     } else if (is.list(obj)) {
-        var_info <- get_vector_var(
-            obj, name, no_preview, page_size, page, sort_by, ascending, filters
+        var_info <- get_list_var(
+            obj, name, no_preview, page_size, page, filters
         )
     } else if (is.matrix(obj) && length(dim(obj)) == 2) {
         var_info <- get_matrix_var(
             obj, name, no_preview, page_size, page, sort_by, ascending, filters
         )
+    } else if (obj_type == 'S4' && !is(obj, "envRefClass")) {
+        var_info <- get_s4_var(obj, name)
+    } else if (obj_type == 'S4' && is(obj, "envRefClass")) {
+        var_info <- get_ref_class_var(obj, name)
     } else {
+        tryCatch({
+            summary <- noquote(toString(obj))
+        }, error=function(e) {
+            sumary <- toString(obj_class)
+        })
+        if (no_preview && nchar(summary) > MAX_SINGLE_VALUE_LENGTH) {
+            summary <- substr(summary, 1, MAX_SINGLE_VALUE_LENGTH)
+        }
```

Comment:
Like I was saying this morning, the `toString` function was failing on S4 classes in this final `else` clause, so just in case any variable type we've left out makes it down this far, I wrapped it in a try-catch